test/controllers/helpers/CheckYourAnswersHelperSpec.scala

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.helpers
+
+import base.SpecBase
+import models.{MaintainProfileAnswers, NiphlNumber, NirmsNumber, UkimsNumber}
+import org.mockito.Mockito.when
+import play.api.i18n.Messages
+import viewmodels.govuk.SummaryListFluency
+import uk.gov.hmrc.govukfrontend.views.Aliases._
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.Key
+import org.scalatestplus.mockito.MockitoSugar.mock
+
+class CheckYourAnswersHelperSpec extends SpecBase with SummaryListFluency {
+
+  val fakeMessages: Messages = mock[Messages]
+
+  when(fakeMessages("site.change")) thenReturn "Change"
+  when(fakeMessages("site.yes")) thenReturn "Yes"
+  when(fakeMessages("site.no")) thenReturn "No"
+
+  "CheckYourAnswersHelper" - {
+
+    val checkYourAnswersHelper = new CheckYourAnswersHelper()
+
+    "createSummaryList" - {
+
+      "must return empty summary list when blank trader goods profile" in {
+        val maintainProfileAnswers: MaintainProfileAnswers = MaintainProfileAnswers()
+        val summaryList                                    = checkYourAnswersHelper.createSummaryList(maintainProfileAnswers)(fakeMessages)
+        summaryList mustBe Seq.empty
+      }
+
+      "must return full summary list when trader goods profile has all fields" in {
+        val expected                                       = List(
+          SummaryListRow(
```

Comment:
Could you extract the creation of these rows to a function so it's easier to read?

Reply:
is this what you mean? - check changes

Reply:
I was thinking more like extracting this bit:

```
def createRow(key: String, value: String, actionItem: String, content: String)
SummaryListRow(
       Key(HtmlContent(key)),
       Value(HtmlContent(value)),
       "",
       Some(
         Actions(
           "",
           List(ActionItem(actionItem, HtmlContent(content), None, "", Map()))
         )
       )
     )
```

Then create a sequence like `Seq(createRow(a,b,c), createRow(d,e,f)...)` but what you did makes it easy to see what is happening so happy with that! Approving PR